Building botan on Windows using nmake:  '_umul128': identifier not found error

I have downloaded the tarball of botan, which has the next version: botan-1.11.28 (tested this too at 1.11.29 - see edit here below). To build the library, i have used the instructions outlined [here](http://botan.randombit.net/manual/building.html#on-windows).

I have issued the following commands on the developer console (Visual Studio 2015 CE 64 bit)
```
D:\build\botan-1.11.28>python configure.py --cc=msvc --prefix="D:\build\botan" --with-debug-info
```

This ran without problems. But when i issue `nmake` after that, I get an error.

```
D:\build\botan-1.11.28>nmake

Microsoft (R) Program Maintenance Utility Version 14.00.23506.0
Copyright (C) Microsoft Corporation.  All rights reserved.

        cl  /MDd /bigobj /DBOTAN_DLL=__declspec(dllexport) /EHs /GR /Zi /FS /O2 /W3 /wd4275 /wd4267 /Ibuild\include /nologo /c src\lib\asn1\alg_id.cpp /Fobuild\obj\lib\asn1_alg_id.obj
alg_id.cpp

        ... other successful cl runs

        cl  /MDd /bigobj /DBOTAN_DLL=__declspec(dllexport) /EHs /GR /Zi /FS /O2 /W3 /wd4275 /wd4267 /Ibuild\include /nologo /c src\lib\mac\poly1305\poly1305.cpp /Fobuild\obj\lib\mac_poly1305.obj
poly1305.cpp
D:\build\botan-1.11.28\build\include\botan/mul128.h(38): warning C4163: '_umul128': not available as an intrinsic function
D:\build\botan-1.11.28\build\include\botan/mul128.h(84): error C3861: '_umul128': identifier not found
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Visual Studio 14.0\VC\BIN\cl.EXE"' : return code '0x2'
Stop.
```
**EDIT 1**: The same problem occurs at the recent build, botan-1.11.29

Is this a known issue ? Is there a way to solve this ?

Oh, an other small question, is the build suitable for both 32 and 64 bit applications or 64 bit only ? It is not clear for me atm. Thank you.

**Appendix:**
Here below is an outline of the output after running the python script (see the first code fragment here above). If you aren't interested in this, feel free to ignore it. It's just for those who wants to get more information.

```
D:\build\botan-1.11.28>python configure.py --cc=msvc --prefix="D:\build\botan" --with-debug-info
   INFO: Platform: OS="Windows" machine="AMD64" proc="Intel64 Family 6 Model 70 Stepping 1, GenuineIntel"
   INFO: Guessing target OS is windows (use --os to set)
   INFO: Guessing target processor is a x86_64/x86_64 (use --cpu to set)
   INFO: Target is msvc-windows-x86_64-x86_64
   INFO: Skipping, by request only - cvc
   INFO: Skipping, dependency failure - sessions_sqlite3
   INFO: Skipping, incompatible CPU - mp_x86_32_msvc
   INFO: Skipping, incompatible OS - beos_stats darwin_secrandom dev_random egd fd_unix proc_walk unix_procs
   INFO: Skipping, incompatible compiler - mp_x86_32 mp_x86_64 simd_altivec
   INFO: Skipping, loaded only if needed by dependency - dyn_load simd_scalar
   INFO: Skipping, requires external dependency - boost bzip2 lzma openssl sqlite3 tpm zlib
   INFO: Using MP module mp_generic
   INFO: Using SIMD module simd_sse2
   INFO: Loading modules adler32 aead aes aes_ni aes_ssse3 aont asn1 auto_rng base base64 bcrypt bigint blake2 block blowfish camellia cascade cast cbc cbc_mac ccm cfb chacha chacha20poly1305 clmul cmac codec_filt comb4p compression crc24 crc32 cryptoapi_rng cryptobox ctr curve25519 datastor des dh dl_algo dl_group dlies dsa eax ec_gfp ec_group ecb ecc_key ecdh ecdsa elgamal eme_oaep eme_pkcs1 eme_raw emsa1 emsa1_bsi emsa_pkcs1 emsa_pssr emsa_raw emsa_x931 entropy ffi filters fpe_fe1 gcm gost_28147 gost_3410 gost_3411 has160 hash hash_id hex hkdf hmac hmac_drbg hmac_rng hres_timer http_util idea idea_sse2 if_algo kasumi kdf kdf1 kdf2 keccak keypair lion locking_allocator mac mars mce mceies md2 md4 md5 mdx_hash mgf1 misty1 mode_pad modes mp mp_generic noekeon noekeon_simd nr numbertheory ocb ofb oid_lookup openpgp par_hash passhash9 pbes2 pbkdf pbkdf1 pbkdf2 pem pk_pad poly1305 prf_tls prf_x942 pubkey rc2 rc4 rc5 rc6 rdrand rdseed rfc3394 rfc6979 rmd128 rmd160 rng rsa rw safer salsa20 seed serpent serpent_simd sessions_sql sha1 sha1_sse2 sha2_32 sha2_64 simd simd_sse2 siphash siv skein srp6 stream system_rng tea threefish threefish_avx2 tiger tls tss twofish utils whirlpool win32_stats x509 x919_mac x931_rng xtea xtea_simd xts
   INFO: Assuming CPU is little endian
   INFO: Assuming unaligned memory access works
   INFO: Using hardlink to link files into build dir (use --link-method to change)
   INFO: Botan 1.11.28 (unreleased undated) build setup is complete
```

--- a/src/build-data/buildh.in
+++ b/src/build-data/buildh.in
@@ -272,4 +272,20 @@ Each read generates 32 bits of output
 // The struct is only declared to force the semicolon, it is never defined.
 #define BOTAN_FORCE_SEMICOLON struct BOTAN_DUMMY_STRUCT
 
+#if defined(BOTAN_TARGET_ARCH_IS_X86_64) && (\
+    (defined(_MSC_VER) && !defined(_WIN64)) || \
+    (defined(__clang__) && !defined(__x86_64__)) || \
+    (defined(__GNUG__) && !defined(__x86_64__)) \
+)
+    #error "Trying to compile Botan configured as x86_64 with non-x86_64 compiler."
+#endif
+
+#if defined(BOTAN_TARGET_ARCH_IS_X86_32) && (\
+    (defined(_MSC_VER) && defined(_WIN64)) || \
+    (defined(__clang__) && !defined(__i386__)) || \
+    (defined(__GNUG__) && !defined(__i386__)) \
+)
+    #error "Trying to compile Botan configured as x86_32 with non-x86_32 compiler."
+#endif
+
 #endif

